### PR TITLE
Remove unnecessary header files

### PR DESCRIPTION
**Pull request by xingwozhonghua126**
“Remove unnecessary header files”

BTW This should be my last pull request for this project, I don’t know what to do next, all the useless includes in the project have been removed by the script.

### CODE REVIEW

**Comment by Vultraz** [comment]
Looks like `desktop/notifications.cpp` at least is missing the definition for `_`.

File: src/display.hpp [hunk 1]
```diff
@@ -43,7 +42,6 @@ class team;
 struct overlay;
 
 namespace halo {
-	class manager;
 }
```
Comment (stevecotton) [inline src/display.hpp:45]: There's a few places like this, where the `namespace x {` and `}` lines could be removed too.

**Comment by stevecotton** [comment]
I generally like this change, but I think it should wait until the end of March before merging. That lets us get 1.18.0 out, and then have a couple of weeks to find out how many bugs get found immediately after the release, before having a change that makes small changes to a huge number of files.

**Comment by Vultraz** [comment]
Could backport.

File: src/ai/composite/component.cpp [hunk 2]
```diff
@@ -19,19 +19,15 @@
...
 #include "log.hpp"
-#include "units/unit.hpp"
 
-#include "ai/formula/ai.hpp"
 
 #include <boost/regex.hpp>
 
 namespace pathfind {
 
-struct pathfind;
```
Comment (CelticMinstrel) [inline src/ai/composite/component.cpp:34]: Here's another one where the namespace can be removed.

File: src/ai/formula/ai.hpp [hunk 3]
```diff
@@ -36,12 +35,10 @@
 #include <vector>                       // for vector
 
 namespace ai { class ai_context; }
-namespace pathfind { struct plain_route; }  // lines 57-57
 struct map_location;
 
 namespace wfl {
 	struct formula_error;
-	class variant;
```
Comment (CelticMinstrel) [inline src/ai/formula/ai.hpp:44]: I'm not sure if it's in scope for this PR, but these two namespace declarations could be merged into one.

File: src/ai/composite/component.cpp (hunk 2 above)
Comment (xingwozhonghua126) [reply src/ai/composite/component.cpp:34]: > Here's another one where the namespace can be removed.

I’m not quite sure what to do, so I won’t proceed with the merger.

File: src/ai/formula/ai.hpp (hunk 3 above)
Comment (xingwozhonghua126) [reply src/ai/formula/ai.hpp:44]: I’m not quite sure what to do, so I won’t proceed with the merger.

Comment (stevecotton) [reply src/ai/formula/ai.hpp:44]: Avoiding any complex changes is good in a PR that touches this many files. It's much easier to review a change that is purely deleting self-contained lines, or self-contained sets of lines.

**Comment by xingwozhonghua126** [comment]
image

@stevecotton 

I'm not sure how to fix this error. Should I not worry about it? I'm going to withdraw for now. Please notify me [@me] if there's something I need to do.

**Comment by stevecotton** [comment]
@xingwozhonghua126 That error is that there's trailing whitespace (a tab character with nothing following it) on that line.

For the two re-added `#include` lines, I'd prefer if you re-added them exactly where they were (so that they'll disappear from the diff if this is squash-merged) instead of moving them into platform-specific `#ifdef`s.

**Comment by stevecotton** [comment]
> Could backport.

Part of the reason for a stable branch is to avoid unnecessary changes which might break, even on platforms that we don't have CI for, so I wouldn't want this to be backported.

@newfrenchy83 has just reported that this missed an indirect include of `<cmath>` into `game_config.cpp` on his dev platform. @newfrenchy83, please would you open a PR adding that file?

**Comment by newfrenchy83** [comment]
> > Could backport.
> 
> Part of the reason for a stable branch is to avoid unnecessary changes which might break, even on platforms that we don't have CI for, so I wouldn't want this to be backported.
> 
> @newfrenchy83 has just reported that this missed an indirect include of `<cmath>` into `game_config.cpp` on his dev platform. @newfrenchy83, please would you open a PR adding that file?

done in https://github.com/wesnoth/wesnoth/pull/8542

**Pull request merged**